central/graphql/resolvers/image_vulnerabilities.go

```diff
@@ -54,8 +55,42 @@ func init() {
 	)
 }
 
+// ImageVulnerabilityResolver represents the supported API on image vulnerabilities
+type ImageVulnerabilityResolver interface {
+	ID(ctx context.Context) graphql.ID
+	CVE(ctx context.Context) string
+	Cvss(ctx context.Context) float64
+	ScoreVersion(ctx context.Context) string
+	Vectors() *EmbeddedVulnerabilityVectorsResolver
+	Link(ctx context.Context) string
+	Summary(ctx context.Context) string
+	FixedByVersion(ctx context.Context) (string, error)
+	IsFixable(ctx context.Context, args RawQuery) (bool, error)
+	LastScanned(ctx context.Context) (*graphql.Time, error)
+	CreatedAt(ctx context.Context) (*graphql.Time, error)
+	DiscoveredAtImage(ctx context.Context, args RawQuery) (*graphql.Time, error)
+	Components(ctx context.Context, args PaginatedQuery) ([]ComponentResolver, error)
+	ComponentCount(ctx context.Context, args RawQuery) (int32, error)
+	Images(ctx context.Context, args PaginatedQuery) ([]*imageResolver, error)
+	ImageCount(ctx context.Context, args RawQuery) (int32, error)
+	Deployments(ctx context.Context, args PaginatedQuery) ([]*deploymentResolver, error)
+	DeploymentCount(ctx context.Context, args RawQuery) (int32, error)
+	EnvImpact(ctx context.Context) (float64, error)
+	Severity(ctx context.Context) string
+	PublishedOn(ctx context.Context) (*graphql.Time, error)
+	LastModified(ctx context.Context) (*graphql.Time, error)
+	ImpactScore(ctx context.Context) float64
+	Suppressed(ctx context.Context) bool
+	SuppressActivation(ctx context.Context) (*graphql.Time, error)
+	SuppressExpiry(ctx context.Context) (*graphql.Time, error)
+	ActiveState(ctx context.Context, args RawQuery) (*activeStateResolver, error)
+	VulnerabilityState(ctx context.Context) string
+	EffectiveVulnerabilityRequest(ctx context.Context) (*VulnerabilityRequestResolver, error)
+	UnusedVarSink(ctx context.Context, args RawQuery) *int32
+}
```

Comment:
This can be broken into:
```
type VulnerabilityCore interface {
        ID(ctx context.Context) graphql.ID
	CVE(ctx context.Context) string
	Cvss(ctx context.Context) float64
	ScoreVersion(ctx context.Context) string
	Vectors() *EmbeddedVulnerabilityVectorsResolver
	Link(ctx context.Context) string
	Summary(ctx context.Context) string
	PublishedOn(ctx context.Context) (*graphql.Time, error)
	CreatedAt(ctx context.Context) (*graphql.Time, error)
	ImpactScore(ctx context.Context) float64
        LastModified(ctx context.Context) (*graphql.Time, error)
       Severity(ctx context.Context) string
       LastScanned(ctx context.Context) (*graphql.Time, error)
         ...
	// core fields.
}

type ImageVulnerabilityResolver interface {
        VulnerabilityMetadata

        // non-core fields
        EnvImpact(ctx context.Context) (float64, error)
        DiscoveredAtImage(ctx context.Context, args RawQuery) (*graphql.Time, error)
 	Suppressed(ctx context.Context) bool
	SuppressActivation(ctx context.Context) (*graphql.Time, error)
	SuppressExpiry(ctx context.Context) (*graphql.Time, error)
	ActiveState(ctx context.Context, args RawQuery) (*activeStateResolver, error)
	VulnerabilityState(ctx context.Context) string
	EffectiveVulnerabilityRequest(ctx context.Context) (*VulnerabilityRequestResolver, error)

        Components(ctx context.Context, args PaginatedQuery) ([]ImageComponentResolver, error)
	ComponentCount(ctx context.Context, args RawQuery) (int32, error)
	Images(ctx context.Context, args PaginatedQuery) ([]*imageResolver, error)
	ImageCount(ctx context.Context, args RawQuery) (int32, error)
	Deployments(ctx context.Context, args PaginatedQuery) ([]*deploymentResolver, error)
	DeploymentCount(ctx context.Context, args RawQuery) (int32, error)
	
	UnusedVarSink(ctx context.Context, args RawQuery) *int32
}

type NodeVulnerabilityResolver interface {
        VulnerabilityMetadata

        EnvImpact(ctx context.Context) (float64, error)
        DiscoveredAtNode(ctx context.Context, args RawQuery) (*graphql.Time, error)
 	Suppressed(ctx context.Context) bool
	SuppressActivation(ctx context.Context) (*graphql.Time, error)
	SuppressExpiry(ctx context.Context) (*graphql.Time, error)
	ActiveState(ctx context.Context, args RawQuery) (*activeStateResolver, error)
	VulnerabilityState(ctx context.Context) string
	EffectiveVulnerabilityRequest(ctx context.Context) (*VulnerabilityRequestResolver, error)

        Components(ctx context.Context, args PaginatedQuery) ([]NodeComponentResolver, error)
	ComponentCount(ctx context.Context, args RawQuery) (int32, error)
	Nodes(ctx context.Context, args PaginatedQuery) ([]*nodeResolver, error)
	NodeCount(ctx context.Context, args RawQuery) (int32, error)
}
```

Reply:
As discussed 1:1, abstracting out the common fields will be addressed in one of the follow up PRs